### PR TITLE
meson: enable more warnings by default

### PR DESCRIPTION
**Pull request by soerengrunewald**
This will enable more warnings by default. As of right now this will not trigger any new warnings.

### CODE REVIEW

File: src/meson.build [hunk 1]
```diff
@@ -18,7 +18,8 @@ if git_exe.found()
     # Get full description based on git tag API version ('v0.2-22-g3d8867c-dirty')
     git_f_api = run_command(
         git_exe, '-C', meson.current_source_dir(),
-            'describe', '--tags', '--always', '--dirty', '--match', 'v*'
+            'describe', '--tags', '--always', '--dirty', '--match', 'v*',
+        check: true
```
Comment (alt-graph) [inline src/meson.build:22]: This "check: true" will probably break the build if you have a `git` executable, but your source code is embedded in some other weird git repo (like vcpkg does it). Note that two lines later we explicitly handle this situation.

**Review by alt-graph — CHANGES_REQUESTED** [review]
I like the warning options, but we should be extremely careful where to put the `run_command` checks. Having git or having the source code in a working copy of our own git repo should not be a prerequisite for building.

File: src/meson.build (hunk 1 above)
Comment (soerengrunewald) [reply src/meson.build]: Good point, I actually did not pay attention to that. Changed!

**Comment by alt-graph** [comment]
Looks good. Thanks, Sören!

File: src/meson.build (hunk 1 above)
Comment (Finii) [reply src/meson.build]: See also

d41fd54 `meson: Allow configuration without working git`

:grimacing:

**Pull request merged**